File: java/Leccion6/Herencia/src/domain/Empleado.java
package domain;

public class Empleado extends Persona {

    private int idEmpleado;
    private double sueldo;
    private static int contadorEmpleados;
    
    public Empleado() {
        this.idEmpleado = ++Empleado.contadorEmpleados;
    }

    public Empleado(String nombre, double sueldo) { // Constructor 2
        //super(nombre);
        this(); // Estamos llamando aquí al constructor vacío
        this.nombre = nombre;
        this.sueldo = sueldo;
    }

    public int getIdEmpleado() {
        return this.idEmpleado;
    }

    public double getSueldo() {
        return this.sueldo;
    }

    public void setSueldo(double sueldo) {
        this.sueldo = sueldo;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Empleado{");
        sb.append("idEmpleado=").append(idEmpleado);
        sb.append(", sueldo=").append(sueldo);
        sb.append(", ").append(super.toString());
        sb.append('}');
        return sb.toString();
    }
    
    
    
    
}

File: Guardini-Philama/JAVASCRIPT/Clase_09/Empleado.js
class Empleado extends Persona {
  static contadorEmpleados = 0;

  //Constructor
  constructor(nombre, apellido, edad, sueldo) {
    super(nombre, apellido, edad); //Atributos heredados de la clase Persona
    this._idEmpleado = ++Empleado.contadorEmpleados; //Atributo propio de la clase
    this.sueldo = sueldo; //Atributo propio de la clase
  }

  //Metodos get y set
  get idEmpleado() {
    return this._idEmpleado;
  }

  get sueldo() {
    return this._sueldo;
  }

  set sueldo(sueldo) {
    this._sueldo = sueldo;
  }

  //Metodo toString
  toString() {
    return `
        ${super.toString()} 
        idEmpleado: ${this._idEmpleado}, 
        sueldo: ${this._sueldo}`;
  }
}
